heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -123,82 +124,44 @@ export default class LintPlugin implements IHeftTaskPlugin {
     // multiple times will only init once.
     await this._ensureInitializedAsync(taskSession, heftConfiguration);
 
-    // Now that we know we have initialized properly, run the linter(s)
-    const lintingPromises: Promise<void>[] = [];
-    if (this._eslintToolPath) {
-      lintingPromises.push(
-        this._runEslintAsync(
-          taskSession,
-          heftConfiguration,
-          this._eslintToolPath,
-          this._eslintConfigFilePath!,
-          tsProgram,
-          changedFiles
-        )
-      );
-    }
-    if (this._tslintToolPath) {
-      lintingPromises.push(
-        this._runTslintAsync(
-          taskSession,
-          heftConfiguration,
-          this._tslintToolPath,
-          this._tslintConfigFilePath!,
-          tsProgram,
-          changedFiles
-        )
-      );
+    const linters: LinterBase<unknown>[] = [];
+    if (this._eslintConfigFilePath && this._eslintToolPath) {
+      const eslintLinter: Eslint = await Eslint.initializeAsync({
+        tsProgram,
+        scopedLogger: taskSession.logger,
+        linterToolPath: this._eslintToolPath,
+        linterConfigFilePath: this._eslintConfigFilePath,
+        buildFolderPath: heftConfiguration.buildFolderPath,
+        buildMetadataFolderPath: taskSession.tempFolderPath
+      });
+      linters.push(eslintLinter);
     }
 
-    await Promise.all(lintingPromises);
-  }
-
-  private async _runEslintAsync(
-    taskSession: IHeftTaskSession,
-    heftConfiguration: HeftConfiguration,
-    eslintToolPath: string,
-    eslintConfigFilePath: string,
-    tsProgram: IExtendedProgram,
-    changedFiles?: ReadonlySet<IExtendedSourceFile> | undefined
-  ): Promise<void> {
-    const eslint: Eslint = new Eslint({
-      scopedLogger: taskSession.logger,
-      eslintPackagePath: eslintToolPath,
-      linterConfigFilePath: eslintConfigFilePath,
-      buildFolderPath: heftConfiguration.buildFolderPath,
-      buildMetadataFolderPath: taskSession.tempFolderPath
-    });
-
-    eslint.printVersionHeader();
+    if (this._tslintConfigFilePath && this._tslintToolPath) {
```

Comment:
At some point, we should consider just dropping support for TSLint.

Reply:
Agree. For now I don't think it hurts since we avoid almost all overhead of loading it, but certainly in future changes to the plugin.